hosts/dialer.go

```diff
@@ -86,3 +91,43 @@ func makeSSHConfig(user string, signer ssh.Signer) (*ssh.ClientConfig, error) {
 
 	return &config, nil
 }
+
+func checkEncryptedKey(sshKey, sshKeyPath string) (ssh.Signer, error) {
+	logrus.Debugf("[ssh] Checking private key")
+	var err error
+	var key ssh.Signer
+	if len(sshKey) > 0 {
+		key, err = parsePrivateKey(sshKey)
+	} else {
+		key, err = parsePrivateKey(privateKeyPath(sshKeyPath))
+	}
+	if err != nil {
```

Comment:
what are we trying to do here? Why do we do all this stuff when there is an error?

Reply:
this check is for encrypted keys, parsing the key with the normal function will generate an error, if this error is encrypted key error we prompt to the user a password to enter, if this generated an error to we return error to the user

Reply:
Then can we make it a bit pretty :) if err == nil - return immediately. Then parse the error